ompi/mca/common/ompio/common_ompio_file_read.c

```diff
@@ -411,88 +449,61 @@ int mca_common_ompio_file_iread (ompio_file_t *fh,
             need_to_copy = true;
         }         
         
-        if ( need_to_copy ) {
-            char *tbuf=NULL;
+        if (need_to_copy) {
+            size_t pipeline_buf_size = OMPIO_MCA_GET(fh, pipeline_buffer_size);
             
-            OMPIO_PREPARE_READ_BUF(fh, buf, count, datatype, tbuf, &ompio_req->req_convertor,
-                                   max_data, 0, decoded_iov, iov_count);
-            
-            ompio_req->req_tbuf = tbuf;
-            ompio_req->req_size = max_data;
+            OMPIO_PREPARE_READ_BUF(fh, buf, count, datatype, ompio_req->req_tbuf,
```

Comment:
This is called for every segment of the pipeline ? The `OMPIO_PREPARE_READ_BUF` constructs and prepare_for_recv the convertor but never cleans it. I don't think it is a safe usage. The convertor should be constructed and prepared only once, and then you use use `set_position` to move it around.

Reply:
yes, but only one at a time, since the code executes one chunk at a time. But we can discuss offline whether I can use set_position instead.